### ISSUE TITLE
Run Dashboards integ test against static endpoint in Jenkins job

### ISSUE THREAD

**Issue by tianleh** [opensearch-project/opensearch-build#1322]
### Is your feature request related to a problem? Please describe

#604

### Describe the solution you'd like

Currently there is only OpenSearch integ test in Jenkins
https://github.com/opensearch-project/opensearch-build/blob/86f9cbe30fb24729466f4a1d33978da3ae6596a6/jenkins/test/testsuite/Jenkinsfile#L63

We want to add such for Dashboards as well. The challenge is how we know the public urls for OS and OSD to feed into test.sh

**Comment by tianleh** [comment]
First need to understand how current OpenSearch integ test runs from Jenkins

**Comment by tianleh** [comment]
<img width="368" alt="Screen Shot 2021-12-13 at 10 53 42 PM" src="https://user-images.githubusercontent.com/60111637/145947922-b768c3a7-51a8-4ac3-8649-92cca0fc51f3.png">

**Comment by tianleh** [comment]
A run needs `opensearch_version`, `build_id`, `architecture`, `platform`.

Follow up question: is this automatically run? If yes, how the parameters are passed? On the other hand, if manual run, then parameters can be passed manually.

**Comment by tianleh** [comment]
One option is once `distribution-build-opensearch-dashboards` finishes, it will trigger another job `dashboards-integ test` and pass the OSD information to it including version, build_id, platform, architecture (the information to concatenate the CI public URL). Then this new job needs to figure out where to get the OS information, especially what is the latest build. This can be done from multiple ways.

1) use the information from OSD run (version, platform, architecture) to look for the S3 bucket of OS and then get latest build_id
2) query the latest/recent successful run of OS jobs and find the one for (version, platform, architecture)

**Comment by tianleh** [comment]
Another option is have a main job to coordinate the build/assemble and then integ test, potentially extending the existing `check-for-build`. If both OS and OSD are triggered, then we can get build_id for both. However, if one of them doesn't trigger, then we need to figure out the latest run with (version, platform, architecture)

**Comment by tianleh** [comment]
DeclarativeJobPropertyTrackerAction.getParameters can only get the input name "INPUT_MANIFEST" but not the actual content of the value.

**Comment by tianleh** [comment]
To summarize current status, I am able to get the latest OpenSearch and OpenSearchDashboards build number. 
http://jenki-jenki-fpgmrv2ryxko-1366042710.us-east-1.elb.amazonaws.com/job/distribution-build-opensearch-dashboards-tianleh/

However, I cannot move further to get the build details. Why is this import? E.g when a OpenSearchDashboards build/assemble is done, it may need to know which build from `distribution-build-opensearch` matches the [version, arch, platform].

**Comment by tianleh** [comment]
@dblock @seraphjiang

**Comment by tianleh** [comment]
To help explain the POC work, I first try to get the latest build number from `distribution-build-opensearch-dashboards` and `distribution-build-opensearch`

This part is working.

Next, assume OpenSearchDashboards is the main driver. We want to

1) get input manifest to understand the version, platform, architecture
2) list OpenSearch recent builds to match the tuple above. (minor release can make the matching logic more complicated)

Currently I am blocked to get the build details (INPUT_MANIFEST) from the build number.

**Comment by dblock** [comment]
I think using the current check-for-build workflow (https://github.com/opensearch-project/opensearch-build/blob/main/jenkins/check-for-build.jenkinsfile) to trigger a new coordinator workflow that runs build and test workflows might be a lot easier. Trying to build something independent that tries to fetch the last good build feels unnecessarily complicated.

**Comment by tianleh** [comment]
After talking with @ohltyler about the PR https://github.com/opensearch-project/opensearch-build/pull/1429, we figure out the following result. Tyler's work is for running **OpenSearch** integ test while mine current task is for such in **OpenSearch Dashboards**.

The OpenSearch Dashboards test job will need to know both OpenSearch and OpenSearch Dashboards public endpoints which is very different from OpenSearch tests which just need to know the OpenSearch public endpoint. To further clarify the difference, since currently OpenSearch and OpenSearch Dashboards distribution build jobs are separate, Tyler's work can be done purely inside the OpenSearch build job while my work could potentially require cross-job information.

I am currently experimenting the "a new coordinator workflow" from https://github.com/opensearch-project/opensearch-build/issues/1322#issuecomment-994793580 which may need to build and test together for both OpenSearch and OpenSearch Dashboards.

To summarize, structure wise there won't be reused work from Tyler's PR towards my work. I would vision "the new coordinator workflow" would trigger the OpenSearch integ test logic by Tyler and also my OpenSearch Dashboards logic in this PR https://github.com/opensearch-project/opensearch-build/pull/1368
(At code level, there may be some references to the manipulation of the artifact rool url.)

CC @ohltyler @dblock @seraphjiang @kavilla @Tengda-He  for visibility

**Comment by dblock** [comment]
I think it may be simpler if Dashboards were treated as a separate standalone product, and just use the latest available OpenSearch build, as needed. Do we really need to coordinate the two?

**Comment by seraphjiang** [comment]
From the issue description, we need public endpoint of both OpenSearch and OpenSearch Dashboards. 
I'm curious how Jenkins code get public endpoint for OpenSearch, it is in environment variable of Jenkins function return? What's difficult to do same for OpenSearch Dashbaords?

**Comment by dblock** [comment]
I see [this code](https://github.com/opensearch-project/opensearch-build/blob/6ed1ce3c583233eae4fe1027969d778cfc7660f7/src/test_workflow/integ_test/integ_test_suite_opensearch_dashboards.py#L71) that passes endpoint information on.

**Comment by tianleh** [comment]
We will run the test against the release endpoint instead of every newly bundled artifact. Found a pre-quest task https://github.com/opensearch-project/opensearch-build/issues/1482

**Comment by tianleh** [comment]
Closed in favor of https://github.com/opensearch-project/opensearch-build/issues/704
